internal/services/containerapps/helpers/container_apps.go

```diff
@@ -0,0 +1,1803 @@
+				},
+
+				"port": {
+					Type:         pluginsdk.TypeInt,
+					Required:     true,
+					ValidateFunc: validation.IntBetween(1, 65535),
+					Description:  "The port number on which to connect. Possible values are between `1` and `65535`.",
+				},
+
+				"host": {
+					Type:        pluginsdk.TypeString,
+					Optional:    true,
+					Description: "The probe hostname. Defaults to the pod IP address. Setting a value for `Host` in `headers` can be used to override this for `http` and `https` type probes.",
+				},
+
+				"path": {
+					Type:        pluginsdk.TypeString,
+					Optional:    true,
+					Computed:    true,
+					Description: "The URI to use with the `host` for http type probes. Not valid for `TCP` type probes. Defaults to `/`.",
+				},
+
+				"header": {
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Elem: &pluginsdk.Resource{
+						Schema: map[string]*pluginsdk.Schema{
+							"name": {
+								Type:         pluginsdk.TypeString,
+								Required:     true,
+								ValidateFunc: validation.StringIsNotEmpty,
+								Description:  "The HTTP Header Name.",
+							},
+
+							"value": {
+								Type:        pluginsdk.TypeString,
+								Required:    true,
+								Description: "The HTTP Header value.",
+							},
+						},
+					},
+				},
+
+				"interval_seconds": {
+					Type:         pluginsdk.TypeInt,
+					Optional:     true,
+					Default:      10,
+					ValidateFunc: validation.IntBetween(1, 240),
+					Description:  "How often, in seconds, the probe should run. Possible values are between `1` and `240`. Defaults to `10`",
+				},
+
+				"timeout": {
+					Type:         pluginsdk.TypeInt,
+					Optional:     true,
+					Default:      1,
+					ValidateFunc: validation.IntBetween(1, 240),
+					Description:  "Time in seconds after which the probe times out. Possible values are between `1` an `240`. Defaults to `1`.",
+				},
+
+				"failure_count_threshold": {
+					Type:         pluginsdk.TypeInt,
+					Optional:     true,
+					Default:      3,
+					ValidateFunc: validation.IntBetween(1, 10),
+					Description:  "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `10`. Defaults to `3`.",
+				},
+
+				"termination_grace_period_seconds": {
+					Type:        pluginsdk.TypeInt,
+					Computed:    true,
+					Description: "The time in seconds after the container is sent the termination signal before the process if forcibly killed.",
+				},
+			},
+		},
+	}
+}
+
+func expandContainerAppStartupProbe(input ContainerAppStartupProbe) containerapps.ContainerAppProbe {
+	probeType := containerapps.TypeStartup
+	result := containerapps.ContainerAppProbe{
+		Type:             &probeType,
+		PeriodSeconds:    pointer.To(int64(input.Interval)),
+		TimeoutSeconds:   pointer.To(int64(input.Timeout)),
+		FailureThreshold: pointer.To(int64(input.FailureThreshold)),
+	}
+
+	switch p := strings.ToUpper(input.Transport); p {
+	case "HTTP", "HTTPS":
```

Comment:
there's constants for this?

Reply:
This is to normalise as the API response may not be in the correct case.